packages/mdc-snackbar/_variables.scss

```diff
@@ -20,9 +20,27 @@
 // THE SOFTWARE.
 //
 
-// Hard coded since the color is not present in any palette.
-$mdc-snackbar-background-color: #323232 !default;
-$mdc-snackbar-foreground-color: white !default;
-// TODO: Better spot to pull this breakpoint?
-//$snackbar-tablet-breakpoint: $grid-tablet-breakpoint;
-$mdc-snackbar-tablet-breakpoint: 600px !default;
+@import "@material/theme/functions";
+@import "@material/theme/variables"; // for mdc-theme-prop-value() function
+@import "@material/theme/color-palette";
+
+$mdc-snackbar-fill-color: mix(mdc-theme-prop-value(on-surface), mdc-theme-prop-value(surface), 80%) !default;
+$mdc-snackbar-label-ink-color: rgba(mdc-theme-prop-value(surface), mdc-theme-text-emphasis(high)) !default;
+$mdc-snackbar-action-icon-ink-color: rgba(mdc-theme-prop-value(surface), mdc-theme-text-emphasis(high)) !default;
+$mdc-snackbar-action-button-ink-color: #bb86fc !default;
```

Comment:
Ideally yes, but our theme system doesn't support that, so Design told me to hard-code it 🤷‍♂️

Reply:
It might at least be more instructive to users looking to tweak variables (and less brittle at least as far as light-on-dark snackbars go) if this were expressed in terms of `lighten($mdc-theme-primary)`... except I can't actually get this color code as a result of it. It's close to 31% but not quite. Did design really give this exact value and if so where did they even get it from??

Reply:
`#bb86fc` comes from the kickoff deck. I don't remember how Design came up with it.

I'm wary of using `lighten()` and `darken()` because they won't work for all colors. We would need a function like `mdc-theme-light-variant()` to ensure proper contrast ratios for accessibility purposes:

https://github.com/material-components/material-components-web/blob/13b560550788b3f991bb876d25ed05342cbddff7/packages/mdc-theme/_functions.scss#L72-L97

However, the `mdc-theme-light-variant()` function was removed in #2473 because it didn't match the guidelines or the other platforms. I don't want to waste our time on something that will just get ripped out.

From my experience, hard-coding the color value is the least controversial option.
Implementing any kind of color logic is just going to open a huge can of worms all over again, which I have no interest in doing—especially when we're trying to get this thing merged before everyone goes on vacation.

If it's really that important to you, let's discuss it offline and create a separate follow up PR so this one doesn't get stuck in limbo forever.